src/utils/codapPhone/index.ts

```diff
@@ -885,12 +886,14 @@ export const getFunctionNames: () => Promise<string[]> = (() => {
       phone.call(
         {
           action: CodapActions.Get,
-          resource: CodapResource.FunctionNames,
+          resource: CodapResource.FormulaEngine,
         },
-        (response: GetFunctionNamesResponse) => {
+        (response: GetFunctionInfoResponse) => {
           if (response.success) {
-            names = response.values;
-            resolve(response.values);
+            const allFunctions: typeof response.values["category"]["fname"][] =
```

Comment:
Would it be worth splitting off the inner type of the response from `FormulaEngine` to simplify the type expression here?

Reply:
Agreed. This should be better.